packages/components/src/internal/components/entities/EntityLineageEditModal.tsx

```diff
@@ -0,0 +1,197 @@
+import React, { FC, memo, useCallback, useEffect, useState } from 'react';
+
+import { AuditBehaviorTypes, Utils } from '@labkey/api';
+
+import {
+    Alert,
+    capitalizeFirstChar,
+    caseInsensitive,
+    createNotification,
+    naturalSort,
+    ParentEntityEditPanel,
+    Progress,
+    QueryModel,
+    resolveErrorMessage,
+    updateRows,
+} from '../../..';
+import { EntityChoice, EntityDataType } from './models';
+import { Button, Modal } from 'react-bootstrap';
+import { getEntityNoun, getUpdatedLineageRowsForBulkEdit } from './utils';
+import { List } from 'immutable';
+import { getOriginalParentsFromSampleLineage, getSampleSelectionLineageData } from '../samples/actions';
+import { ParentEntityLineageColumns } from './constants';
+
+interface Props {
+    queryModel: QueryModel;
+    onCancel: () => void;
+    onSuccess: () => void;
+    childEntityDataType: EntityDataType;
+    auditBehavior?: AuditBehaviorTypes;
+    parentEntityDataTypes: EntityDataType[];
+    parentNounPlural: string;
+    parentNounSingular: string;
+}
+
+export const EntityLineageEditModal: FC<Props> = memo(props => {
+    const {
+        auditBehavior,
+        queryModel,
+        onCancel,
+        childEntityDataType,
+        onSuccess,
+        parentEntityDataTypes,
+        parentNounPlural,
+        parentNounSingular,
+    } = props;
+    const [submitting, setSubmitting] = useState(false);
+    const [numAliquots, setNumAliquots] = useState<number>(undefined);
+    const [nonAliquots, setNonAliquots] = useState<Record<string, any>>(undefined);
+    const [errorMessage, setErrorMessage] = useState(undefined);
+    const [hasParentUpdates, setHasParentUpdates] = useState<boolean>(false);
+    const lcParentNounPlural = parentNounPlural.toLowerCase();
+    const [selectedParents, setSelectedParents] = useState<List<EntityChoice>>(List<EntityChoice>());
+
+    useEffect(() => {
+        getSampleSelectionLineageData(List.of(...queryModel.selections), queryModel.queryName, List.of('RowId', 'Name', 'LSID', 'IsAliquot').concat(ParentEntityLineageColumns).toArray())
+            .then(async response => {
+                const { key, models } = response;
+                const nonAliquots = {};
+                let aliquotCount = 0;
+                Object.keys(models[key]).forEach(id => {
+                    const d = models[key][id];
+                    if (caseInsensitive(d, 'IsAliquot')['value']) {
+                        aliquotCount++;
+                    } else {
+                        nonAliquots[id] = d;
+                    }
+                });
+                setNumAliquots(aliquotCount);
+                setNonAliquots(nonAliquots);
+
+            })
+            .catch(error => {
+                setErrorMessage(error);
+            });
+    }, [])
+
+
+    const onParentChange = useCallback((entityParents: List<EntityChoice>) => {
+        setSelectedParents(entityParents);
+        setHasParentUpdates(entityParents.size > 0);
```

Comment:
does this need to be stored as a separate state variable or could `hasParentUpdates` just be calculated as needed based on the `selectedParents` state variable?

Reply:
no probably not. It is likely very safe in this case with just a single usage, but I always worry a bit about them getting out of sync.